src/main/java/com/fasterxml/jackson/databind/type/TypeFactory.java

```diff
@@ -480,23 +480,82 @@ private TypeBindings _bindingsForSubtype(JavaType baseType, int typeParamCount,
         // (hopefully passing null Class for root is ok)
         int baseCount = baseType.containedTypeCount();
         if (baseCount == typeParamCount) {
-            if (typeParamCount == 1) {
```

Comment:
I split the difference by removing the two inlined cases and using the more optimal array-based invocation for the general case.  Just did this for code clarity.